acs5zippoverty.py

```diff
@@ -0,0 +1,59 @@
+import json
+from census_response import getCensusResponse
+
+#Estimate!!Total!!Population for whom poverty status is determined
+pop_table = 'S1701_C01_001E'
+#Estimate!!Below poverty level!!Population for whom poverty status is determined
+poverty_table = 'S1701_C02_001E'
+get_ls = [pop_table, poverty_table]
+
+subject_table = 'https://api.census.gov/data/2018/acs/acs5/subject?'
+
+#zip level request 
+#Unable to filter request by state
+geo = 'zip code tabulation area:*'
+
+response = getCensusResponse(subject_table, get_ls, geo)
+
+#poverty_data is list of lists, where list elements are rows
+#header row: name, population total, population poverty, fip state, fip county
+poverty_data = response.json()
+
+#filter zipcodes
+#IL zipcodes begin with numbers in zip_ls_IL
+zip_ls_IL = ['60', '61', '62']
+
+poverty_data_IL = []
+
+for d in poverty_data[1:]:
+    #print(d[3][:2])
+    if d[3][:2] in zip_ls_IL:
+        poverty_data_IL.append(d)
+        
+#Json format
+#{'countyfip': (countyfip = concatenation state and county fip numbers)
+    #{'metric_one': 1234,
+    #'metric_two': 5678}}
+    
+final_json = {}
+
+for d in poverty_data_IL:
+    #set variables to list elements
+    name, pop_total_str, pop_poverty_str, zipcode = d
+    #convert strings to ints
+    pop_total_int = int(pop_total_str)
+    pop_poverty_int = int(pop_poverty_str)
+    #calculate percent poverty
+    if pop_total_int != 0:
+        pct_poverty = pop_poverty_int / pop_total_int * 100
+    else:
+        pct_poverty = 'Undefined'
+    #create zip json
+    zip_json = {'name_zip': name, 'population_total': pop_total_int, 'population_poverty': pop_poverty_int, 'percent_poverty': pct_poverty}
+    #set county key to county json value
+    final_json[zipcode] = zip_json
+
+#save file
+with open('final_jsons/acs5zippoverty_output.json', 'w') as f:
```

Comment:
Hey this PR is looking really good! As a last step, could you write a test that does the following? 

1. Runs this script
2. Reads in `final_jsons/acs5zippoverty_output.json` 
3. Asserts that this json has the required format and types ie that it matches
```
#Json format
#{'countyfip': (countyfip = concatenation state and county fip numbers)
    #{'metric_one': 1234,
    #'metric_two': 5678}}
```

Reply:
I made a test file, but I am not familiar enough with the pytest framework to debug some of the errors it is raising for me.

Reply:
Ok sounds good! I’ll pull down this branch and take a look. How are you running the tests?

Reply:
Ah ok this is a simple fix -- you just need to rename the functions to match how pytest discovers tests. When I did that the tests passed 🎉  I can explain more in a zoom call!